core/src/main/java/be/isach/ultracosmetics/hook/PEconomy.java

```diff
@@ -0,0 +1,33 @@
+package be.isach.ultracosmetics.economy;
+
+import com.github.SoKnight.PEconomy.API.PEconomyAPI;
+import org.bukkit.entity.Player;
+
+public class PEconomy implements EconomyManager {
+
+    private PEconomyAPI pEconomyAPI;
+
+    public PEconomy() {
+        this.pEconomyAPI = PEconomyAPI.getPEconomyAPI();
+    }
+
+    @Override
+    public double getBalance(Player player) {
+        return pEconomyAPI.getBalance(player.getUniqueId());
+    }
+
+    @Override
+    public boolean has(Player player, double amount) {
+        return pEconomyAPI.getBalance(player.getUniqueId()) >= amount;
+    }
+
+    @Override
+    public void withdraw(Player player, double amount) {
+        pEconomyAPI.withdraw(player.getUniqueId(), amount);
+    }
+
+    @Override
+    public void deposit(Player player, double amount) {
+        pEconomyAPI.deposit(player.getUniqueId(), amount);
+    }
```

Comment:
It looks like none of these methods actually exist? Looking at the API [here](https://github.com/SoKnight/PEconomy/blob/master/src/main/java/ru/soknight/peconomy/api/PEconomyAPI.java)

It also looks like the API methods need to be run on a separate thread since they interact with a database, so I'll probably have to do something fancy like Treasury has. I'll just close this for now